halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/deck/ApacheSpinnakerProfileFactory.java

```diff
@@ -77,13 +78,24 @@ protected void setProfile(
 
   @Override
   protected Map<String, Object> getBindings(
-      DeploymentConfiguration deploymentConfiguration, SpinnakerRuntimeSettings endpoints) {
+      DeploymentConfiguration deploymentConfiguration,
+      Profile profile,
+      SpinnakerRuntimeSettings endpoints) {
     TemplatedResource resource = new StringResource(SSL_TEMPLATE);
     Map<String, Object> bindings = new HashMap<>();
+    BindingsSecretDecrypter bsc =
```

Comment:
Creating your service dependencies with `new` is generally frowned upon. Instead, you should have Spring inject your dependencies.

You probably want a `@Component BindingsSecretDecrypterFactory` with a `SecretSessionManager` parameter and a `createDecrypter(Profile profile, String decryptedOutputDirectory)` method. Then inject that factory into an `@Autowired` constructor of this class...